### PR TITLE
Fix mathematical correctness and align behavior for average op

### LINKED ISSUE #23250: Mathematical incorrectness and inconsistent behavior in keras.ops.average across backends
The keras.ops.average (and the underlying Average op) implementation deviates from numpy.average behavior under specific scenarios involving weights, sequence axes (tuple or list), and 1D weight reduction. This leads to crashes, broadcasting errors, or incorrect mathematical results depending on the active backend.

1. PyTorch Backend
Denominator Dimension Misalignment (axis=None): The denominator Formulation has a hardcoded dim=-1, which causes a dimension/shape mismatch or incorrect reduction when axis=None or when reducing along multiple dimensions with weights.
1D Weights Broadcasting Failure: 1D weights fail to broadcast because they always try to match the last dimension of the input, ignoring the actual axis you are trying to average.

2. TensorFlow Backend
Crash on List-type Axis: Passing a list as the axis parameter (e.g., axis=[1]) crashes in _rank_not_equal_case because element access assumes a type that supports explicit indexing logic but fails on certain sequence permutations.

3. OpenVINO Backend
Mathematical Error: The weighted average formula incorrectly uses reduce_mean instead of reduce_sum in the denominator formulation, scaling the divisor incorrectly and returning mathematically wrong averages for non-uniform weights.
Broadcasting Logic: Shares/mirror the same 1D weight broadcasting misalignment as the PyTorch backend.

4. Core Ops / Shape Inference
Sequence/Tuple Axis Ignored: Average.compute_output_spec lacks handling for tuple-based reduction axes when applying weights, failing to reduce multiple dimensions in the symbolic shape inference logic.

Bug reproduction notebook: 
https://colab.research.google.com/drive/17Upp_Vs0P9W3-BftoksML-sK7ic05i00?usp=sharing

### PR DESCRIPTION
**Pull request by buildwithsuhana**
Fixes https://github.com/keras-team/keras/issues/23250

## Description
This PR aligns the behavior of keras.ops.average across all backends (TensorFlow, Torch, JAX, NumPy, and OpenVINO) with NumPy semantics and ensures mathematical correctness.

Key Changes:

1. Support for Sequence Axes: Enabled custom reductions on multiple dims by fixing symbolic dimension dropping in Average.compute_output_spec when applied with sequence axes.

2. PyTorch Backend (backend/torch/numpy.py)
Flexible Reductions Divisor: Stopped hardcoding dim=-1 in the weighted sum divisor loop. This fixes errors when reducing globally (axis=None) or on a custom axis.
Internal Axis Broadcasting: 1D weight arrays will now broadcast properly across intermediary axes in higher-rank tensors instead of unconditionally aligning to the right (failing Shape alignment).

3. OpenVINO Backend (backend/openvino/numpy.py)
Formula Correctness: Switched reduce_mean to reduce_sum in the denominator logic to ensure mathematically accurate weighted averages.
Vector Shape Alignment: Patched the broadcasting sequence logic to allow 1D vectors to act alongside chosen non-final dimensions.

4. TensorFlow Backend (backend/tensorflow/numpy.py)
Mutable Sequence Support: Safeguard _rank_not_equal_case from crashing when list-type axes ([0] instead of (0,)) are passed.
## Contributor Agreement

**Please review our [AI-Assisted Contribution Policy](../CONTRIBUTING.md#ai-assisted-contribution-policy) and check all boxes below before submitting your PR for review:**

- [x] I am a human, and not a bot.
- [x] I will be responsible for responding to review comments in a timely manner.
- [x] I will work with the maintainers to push this PR forward until submission.

**Note:** Failing to adhere to this agreement may result in your future PRs no longer being reviewed.

### CODE REVIEW

File: keras/src/ops/numpy.py [hunk 1]
```diff
@@ -1391,7 +1400,28 @@ def average(x, axis=None, weights=None):
...
+                )
+            axis_val = axis[0] if isinstance(axis, (list, tuple)) else axis
+            axis_val = canonicalize_axis(axis_val, len(x_shape))
+            if weights_shape[0] != x_shape[axis_val]:
+                raise ValueError(
+                    "Length of 1D weights must match the dimension of the "
+                    f"axis. Received: weights.shape[0]={weights_shape[0]}, "
+                    "whereas the dimension of the axis is "
+                    f"x.shape[{axis_val}]={x_shape[axis_val]}."
+                )
```
Comment (hertschuh) [inline keras/src/ops/numpy.py]: Unfortunalely, this code cannot be here. It has to be in the backend specific implementation.

The code here can only be:
```python
if any_symbolic_tensors(...):
    return ...
return backend.numpy...
```

One reason is that the internal use of the ops within keras bypasses this code and goes straight to the backend implementation.

So please test all the backends and add this to the backends that need it. You might need to not check the error message in your unit tests and only the fact that an exception is raised.

**Review by hertschuh — CHANGES_REQUESTED** [review]
Thanks for catching this!

File: keras/src/ops/numpy.py (hunk 1 above)
Comment (buildwithsuhana) [reply keras/src/ops/numpy.py]: Done.

**Review by hertschuh — APPROVED** [review]
Thanks for the fix!
